libsignal-service/src/account_manager.rs

```diff
@@ -1,15 +1,25 @@
+use crate::pre_keys::{PreKeyEntity, PreKeyState};
 use crate::push_service::{
     ConfirmDeviceMessage, DeviceId, PushService, ServiceError,
     SmsVerificationCodeResponse, VoiceVerificationCodeResponse,
 };
 
+use std::convert::TryFrom;
+use std::time::SystemTime;
+
+use libsignal_protocol::{Context, StoreContext};
+
 pub struct AccountManager<Service> {
+    context: Context,
     service: Service,
 }
 
+const PRE_KEY_MINIMUM: u32 = 10;
+const PRE_KEY_BATCH_SIZE: u32 = 100;
+
 impl<Service: PushService> AccountManager<Service> {
-    pub fn new(service: Service) -> Self {
-        Self { service }
+    pub fn new(context: Context, service: Service) -> Self {
+        Self { service, context }
```

Comment:
being pedantic here, but maybe swap the order of the fields to match the constructor?

Reply:
I was wondering whether the `StoreContext` from below should maybe also go in the constructor.